editor/animation_track_editor.cpp

```diff
@@ -5982,6 +5982,56 @@ void AnimationTrackEditor::_edit_menu_pressed(int p_option) {
 			undo_redo->commit_action();
 		} break;
 
+		case EDIT_SET_START_OFFSET: {
+			EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+			undo_redo->create_action(TTR("Animation Set Start Offset"), UndoRedo::MERGE_ENDS);
+			for (const KeyValue<SelectedKey, KeyInfo> &E : selection) {
+				if (animation->track_get_type(E.key.track) != Animation::TYPE_AUDIO) {
+					continue;
+				}
+				Ref<AudioStream> stream = animation->audio_track_get_key_stream(E.key.track, E.key.key);
+				real_t len = stream->get_length() - animation->audio_track_get_key_end_offset(E.key.track, E.key.key);
```

Comment:
I see you're using `real_t` throughout this code, is there a specific reason for that?

AFAIK these APIs like `AudioStream::get_length()` return `double`. It doesn't seem useful to cast those to `float` in `precision=single` builds (which is the default).

As a reminder, `real_t` is only for things like vector types whose precision can be configured for performance purposes in heavy computational stuff like transforms/physics/rendering.

For audio/animation I don't think `real_t` is a good choice, anything that's `Variant::FLOAT` in the API should use `double` in C++.

Reply:
I remember that it was preferable for keys not to be double at least when inserting keys, as there is a possibility of duplicate keys due to errors between float and dloble, which are not visible in the editor.

But for now, I changed them to respect the retrieved type.